transports/dns/src/lib.rs

```diff
@@ -170,7 +176,56 @@ where
                             })
                             .next()
                             .ok_or_else(|| DnsErr::ResolveFail(name))
-                    }.left_future()
+                    }.boxed().left_future()
```

Comment:
You normally don't have to box the future. Just put an `async` block around the entire code instead of having two distinct `async` blocks.

Since this is a lot of code, it might be worth moving it to a separate freestanding function.